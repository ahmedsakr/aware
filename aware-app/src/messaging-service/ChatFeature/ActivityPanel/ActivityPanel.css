#activity-panel {
    display: flex;
    align-items: center;
    text-align: left;
    background-color: #ebebeb;
    height: 5%;
    padding-left: 5px;
}