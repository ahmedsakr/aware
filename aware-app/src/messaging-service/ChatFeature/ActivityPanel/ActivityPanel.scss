$background_color: #ebebeb;

#activity-panel {
    display: flex;
    flex: 0 0 auto;
    align-items: center;
    text-align: left;
    background-color: $background_color;
    height: 5%;
    padding-left: 5px;
}