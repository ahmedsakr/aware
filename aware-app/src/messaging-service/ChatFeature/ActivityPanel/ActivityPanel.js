import React, { Component } from 'react';
import './ActivityPanel.css'
import ProfilePicture from '../Profile/ProfilePicture';

class ActivityPanel extends Component {

    render() {
        return(
            <div class="col-sm-12" id="activity-panel">
                <ProfilePicture instance="activity" activity="online"  picture="ahmed-pic.jpg"/>
                <ProfilePicture instance="activity" activity="online" picture="josh-pic.jpg"/>
                <ProfilePicture instance="activity" activity="online" picture="louis-ck-pic.JPG"/>
                <ProfilePicture instance="activity" activity="online" picture="mia-khalifa-pic.jpg"/>
                <ProfilePicture instance="activity" activity="offline" picture="bill-gates-pic.jpg"/>
                <ProfilePicture instance="activity" activity="offline" picture="megan-fox-pic.jpg"/>
            </div>
        );
    }
}

export default ActivityPanel;
