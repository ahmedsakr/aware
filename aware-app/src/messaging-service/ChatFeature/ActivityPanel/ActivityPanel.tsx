import React from 'react';
import './ActivityPanel.scss'
import ProfilePicture from '../Profile/ProfilePicture';

type ActivityPanelProps = {
    socket: SocketIOClient.Socket,
    activeRoom: string
};

type ActivityPanelState = {
    activityPanelUsers: []
};

export default class ActivityPanel extends React.Component<ActivityPanelProps, ActivityPanelState> {
    constructor(props: ActivityPanelProps) {
        super(props);

        this.state = {
            activityPanelUsers: []
        }
    }

    componentWillMount() {
        if (this.props.socket) {
            this.props.socket.on('active users', (users: []) => {
                this.setState({
                    activityPanelUsers: users
                })
            })
        }
    }

    componentDidUpdate() {
        if (this.props.socket) {
            this.props.socket.emit('active users', this.props.activeRoom);
        }
    }

    shouldComponentUpdate(nextProps: ActivityPanelProps, nextState: ActivityPanelState): boolean {
        return this.props.activeRoom !== nextProps.activeRoom;
    }

    render(): JSX.Element {
        return (
            <div className="col-sm-12" id="activity-panel">
                {
                    this.state.activityPanelUsers.map((user: any) => {
                        return (
                            <ProfilePicture instance="activity" activity="online" picture={user.username + "-pic.jpg"} />
                        )
                    })
                }
            </div>
        );
    }
}
