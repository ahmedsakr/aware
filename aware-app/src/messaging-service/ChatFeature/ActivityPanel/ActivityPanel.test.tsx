import React from 'react';
import { shallow } from 'enzyme';

import ActivityPanel from './ActivityPanel';

describe('ActivityPanel', () => {
    it('should render correctly', () => {

        const component = shallow(<ActivityPanel />);

        expect(component).toMatchSnapshot();
    });
});
