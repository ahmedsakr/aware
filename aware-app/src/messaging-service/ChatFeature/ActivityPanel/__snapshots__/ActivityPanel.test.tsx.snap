version https://git-lfs.github.com/spec/v1
oid sha256:e73eb74120237dc1330274ff4416da2dcebc0c756c31a9dc0a5159b1e298a280
size 763
